app/src/main/java/org/thoughtcrime/securesms/onboarding/LandingActivity.kt

```diff
@@ -19,28 +53,128 @@ class LandingActivity : BaseActionBarActivity() {
         setUpActionBarSessionLogo(true)
-        with(binding) {
-            fakeChatView.startAnimating()
-            registerButton.setOnClickListener { register() }
-            restoreButton.setOnClickListener { link() }
-            linkButton.setOnClickListener { link() }
-        }
+
+        ComposeView(this)
+            .apply { setContent { AppTheme { LandingScreen() } } }
+            .let(::setContentView)
+
         IdentityKeyUtil.generateIdentityKeyPair(this)
         TextSecurePreferences.setPasswordDisabled(this, true)
         // AC: This is a temporary workaround to trick the old code that the screen is unlocked.
         KeyCachingService.setMasterSecret(applicationContext, Object())
     }
 
-    private fun register() {
-        val intent = Intent(this, RegisterActivity::class.java)
-        push(intent)
+    @Preview
+    @Composable
+    private fun LandingScreen(
+        @PreviewParameter(ThemeResPreviewParameterProvider::class) themeResId: Int
+    ) {
+        PreviewTheme(themeResId) {
+            LandingScreen()
+        }
+    }
+
+    @Composable
+    private fun LandingScreen() {
+        Column(modifier = Modifier.padding(horizontal = 36.dp)) {
+            Spacer(modifier = Modifier.weight(1f))
+            Text(stringResource(R.string.onboardingBubblePrivacyInYourPocket), modifier = Modifier.align(Alignment.CenterHorizontally), style = MaterialTheme.typography.h4, textAlign = TextAlign.Center)
+            Spacer(modifier = Modifier.height(24.dp))
+            IncomingText(stringResource(R.string.onboardingBubbleWelcomeToSession))
+            Spacer(modifier = Modifier.height(14.dp))
+            OutgoingText(stringResource(R.string.onboardingBubbleSessionIsEngineered))
+            Spacer(modifier = Modifier.height(14.dp))
+            IncomingText(stringResource(R.string.onboardingBubbleNoPhoneNumber))
+            Spacer(modifier = Modifier.height(14.dp))
+            OutgoingText(stringResource(R.string.onboardingBubbleCreatingAnAccountIsEasy))
+            Spacer(modifier = Modifier.weight(1f))
+
+            OutlineButton(
+                text = stringResource(R.string.onboardingAccountCreate),
+                modifier = Modifier
+                    .width(262.dp)
+                    .align(Alignment.CenterHorizontally),
+                contentDescription = GetString(R.string.AccessibilityId_create_account_button)
+            ) { startPickDisplayNameActivity() }
+            Spacer(modifier = Modifier.height(14.dp))
+            FilledButton(
+                text = stringResource(R.string.onboardingAccountExists),
+                modifier = Modifier
+                    .width(262.dp)
+                    .align(Alignment.CenterHorizontally),
+                contentDescription = GetString(R.string.AccessibilityId_restore_account_button)
+            ) { startLinkDeviceActivity() }
+            Spacer(modifier = Modifier.height(8.dp))
+            BorderlessButton(
```

Comment:
There's a lot of very specific values hardcoded here ( the width, the font size, the lineheight). Is this button reused anywhere? Shouldn't it be made as a reusable component?
Also all the values regarding typography should be extracted into an organised Type file and accessed appropriately instead of hardcoded each time into every composables.
We might need to do an overall styling pass on all our composables later, because I think it has  been done everywhere so far.

Reply:
part of onboarding